server/app/actions/ApplicantAuthorizationAction.java

```diff
@@ -75,7 +75,10 @@ private long getApplicantId(Request request, RouteExtractor routeExtractor) {
 
   /** Check if the profile is authorized to access the applicant's data. */
   private CompletionStage<Void> checkApplicantAuthorization(Request request, long applicantId) {
-    return profileUtils.currentUserProfile(request).orElseThrow().checkAuthorization(applicantId);
+    return profileUtils
+        .optionalCurrentUserProfile(request)
```

Comment:
**suggestion:** Should we replace this with `currentUserProfile(request)` and that will do the `orElseThrow` within the function? Here and in other cases below?

Reply:
Yes, working on that in the next PR :)